loading.bs

```diff
@@ -331,6 +376,322 @@ Note: Applying the signed exchange's response here has the effect of letting a
+    [=exchange=] with the URL of the <{link/rel/allowed-alt-sxg}> link's [=Link
+    Target=] URL and the result of [=response/cloning=] the inner response of
+    the [=alternate signed exchange=], and store the exchange to the
+    [=Document/prefetched subresource signed exchanges=] of the document.
+
+</div>
+
+<h3 algorithm id="mp-document">Monkeypatch Document</h3>
+
+At the end of
+[the Document object](https://html.spec.whatwg.org/multipage/dom.html#the-document-object)
+section, add the following lines:
+
+<blockquote dfn-for="Document">
+
+The {{Document}} has a set of <dfn dfn>prefetched signed exchanges for
+navigation</dfn>, which is a set of [=exchanges=], initially empty.  <span
+class="note">This might merge with the set of prefetched resources when
+<{link/rel/prefetch}> is specified more completely.</span>
+
+The {{Document}} has a set of <dfn dfn>prefetched
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+The {{Document}} has a set of <dfn dfn>transferred
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+The {{Document}} has a set of <dfn dfn>allowed
+alternate subresource signed exchange links</dfn>, which is a set of
+<{link/rel/allowed-alt-sxg}> links, initially empty.
+
+</blockquote>
+
+<h3 algorithm id="mp-navigation-params">Monkeypatch navigation params struct</h3>
+
+In the [=navigation params=], add the following items:
+
+<dl dfn-for="navigation params">
+
+: <dfn>prefetched subresource signed exchanges</dfn>
+:: A set of [=exchanges=], initially empty.
+: <dfn>allowed alternate subresource signed exchange links</dfn>
+:: A set of <{link/rel/allowed-alt-sxg}> links, initially empty.
+
+</dl>
+
+<h3 algorithm id="mp-fetching-the-resource-hint-link">Monkeypatch Fetching the resource hint link</h3>
+
+At the end of the [Fetching the resource hint
+link](https://w3c.github.io/resource-hints/#fetching-the-resource-hint-link)
+section in [[resource-hints]], add the following lines:
+
+The user agent SHOULD run the following steps if the response of a
+<{link/rel/prefetch}>ed resource [=response/came from a signed exchange=]:
+
+    1. Let |clonedResponse| be the result of [=response/cloning=] the response.
+    1. Create an [=exchange=] with the request URL and |clonedResponse|,
+        and store the exchange to the
+        [=Document/prefetched signed exchanges for navigation=] of the Document.
+
+<h3 algorithm id="mp-process-a-navigate-fetch">
+Monkeypatch process a navigate fetch</h3>
+In [process a navigate fetch](https://html.spec.whatwg.org/multipage/browsing-the-web.html#process-a-navigate-fetch)
+before
+
+> 9. While true:
+
+add the following steps:
+
+    9. Let |allowed alternate subresource signed exchange links| be null.
+
+    10. If |sourceBrowsingContext|'s [=active document=]'s
+        [=Document/prefetched signed exchanges for navigation=] has a matching
```

Comment:
"has a matching exchange" is a bit handwavy. Might be better to define "prefetched signed exchanged for navigation" as a [map](https://infra.spec.whatwg.org/#maps)